File: lib/src/main/java/by/mrrockka/service/FinalePlacesService.java
package by.mrrockka.service;

import by.mrrockka.domain.finaleplaces.FinalePlaces;
import by.mrrockka.mapper.FinalePlacesMapper;
import by.mrrockka.repo.finalplaces.FinalePlacesRepository;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.UUID;

@Service
@RequiredArgsConstructor
public class FinalePlacesService {

  private final FinalePlacesRepository finalePlacesRepository;
  private final FinalePlacesMapper finalePlacesMapper;

  public FinalePlaces getByGameId(@NonNull final UUID gameId) {
    return finalePlacesRepository.findByGameId(gameId)
      .map(finalePlacesMapper::toDomain)
      .orElse(null);
  }

  public void store(@NonNull final UUID gameId, @NonNull final FinalePlaces finalePlaces) {
    finalePlacesRepository.save(finalePlacesMapper.toEntity(gameId, finalePlaces));
  }
}

File: core/src/main/java/by/mrrockka/service/FinalePlacesService.kt
package by.mrrockka.service

import by.mrrockka.domain.finaleplaces.FinalePlaces
import by.mrrockka.mapper.FinalePlacesMapper
import by.mrrockka.repo.finalplaces.FinalePlacesRepository
import lombok.RequiredArgsConstructor
import org.springframework.stereotype.Service
import java.util.*

@Service
@RequiredArgsConstructor
class FinalePlacesService {
    private lateinit var finalePlacesRepository: FinalePlacesRepository
    private lateinit var finalePlacesMapper: FinalePlacesMapper

    //todo: refactor optionals
    fun getByGameId(gameId: UUID): FinalePlaces? {
        return finalePlacesRepository.findByGameId(gameId)
                .map { entity -> finalePlacesMapper.toDomain(entity) }
                .orElse(null)
    }

    fun getAllByPersonId(personId: UUID): MutableList<FinalePlaces> {
        return finalePlacesRepository.findAllByPersonId(personId).stream()
                .map { entity -> finalePlacesMapper.toDomain(entity) }
                .toList()
    }

    fun store(gameId: UUID, finalePlaces: FinalePlaces) {
        finalePlacesRepository.save(finalePlacesMapper.toEntity(gameId, finalePlaces))
    }
}
